index.js

```diff
@@ -0,0 +1,101 @@
+// NOTE: This was my first attempt at the project. After this I tried to refactor my code in index_2.js, which works up to the point where I make a post request to the api (I can't find a way to stop it from going to the api url).
+
+$(document).ready(function () {
+  // click to see all the trips
+  $('#get-trips').on('click', () => {
+    // make api request to get all of the names of the trips
+    $.get('https://trektravel.herokuapp.com/trips', function(data) {
+      // iterate though the api resonse and add all the trip names as HTML h5 to the page
+      for(i = 0; i < data.length; i++) {
+        let id = data[i]["id"];
+        let name = data[i]["name"];
+
+        // make a jQuery object from the api response
+        let tripHTML = $('<article class="row">' + `<h3 class="${id} large-9 column">` + name + '</h3>' + `<button id="button${id}">Trip details</button>` + '</article>');
+        // hide the 'see all trips' button
+        $('#get-trips').hide()
+        // append the trip names to the all-trips section
+        $('#all-trips').append(tripHTML);
+
+        // create click event for each trip name
+        tripHTML.on('click', 'button', (event) => {
+
+          let url =  'https://trektravel.herokuapp.com/trips/' + id
+
+          // make the api request to get the details for the trip
+          $.get(url, function(event) {
+            let continent = event["continent"];
+            let weeks = event["weeks"];
+            let about = event["about"];
+            let category = event["category"];
+            let cost = event["cost"];
+
+            let tripDetails = $(
+            '<ul>' +
+            '<li>' + 'ID: ' + id + '</li>' +
+            '<li>' + 'Continent: ' + continent + '</li>' +
+            '<li>' + 'Category: ' + category + '</li>' +
+            '<li>' + 'Number of weeks: ' + weeks + '</li>' +
+            '<li>' + 'Cost: $' + cost + '</li>' +
+            '</ul>' +
+            '<p>' + 'Trip details: ' + about + '</p>' +
+            `<button id="trip${id}">Book this trip</button>`);
+
+            // append the trip details to the trip name (above I added the 'id' as a class for the h5 containing each name)
+            $(`.${id}`).parent().append(tripDetails);
+
+            // create a click event for the button to book a trip
+            $(`#trip${id}`).click((event) => {
+              // stops the tripHTML click event from running
+              event.stopPropagation();
```

Comment:
This is a callback inside a callback inside a callback inside a callback inside a callback! To me, this is a good indicator that you should break some of these out into separate named functions, defined at the top of the file. Each jQuery event handler and AJAX success or failure callback should get its own function.

See also [callback hell](http://callbackhell.com/)